src/js/components/Typeahead.js

```diff
@@ -0,0 +1,73 @@
+import InnerTypeahead from 'mesosphere-react-typeahead';
+/* eslint-disable no-unused-vars */
+import React, {PropTypes} from 'react';
+/* eslint-enable no-unused-vars */
+
+import FilterInputText from './FilterInputText';
+
+class Typeahead extends FilterInputText {
+  handleInputChange(input) {
+    this.props.handleFilterChange(input);
+  }
+
+  componentWillUpdate() {
+    if (this.inputField == null && this.typeahead) {
+      if (this.typeahead.refs.inner) {
+        if (this.typeahead.refs.inner.inputComponent) {
+          this.inputField = this.typeahead.refs.inner.inputComponent.input;
+        }
+      }
+    }
+  }
+
+  renderClearIcon() {
+    if (this.inputField && this.inputField.value) {
```

Comment:
if `value == 0` we wouldn't be able to clear it right? I think we should make this `this.inputField.value != null`.